### PR TITLE
feat: introduce postgres dbstore with migrations

### PR DESCRIPTION
**Pull request by mfiedorowicz**
- add postgres db to docker compose
- add postgres schemas/migrations and initial queries
- generate dbstore code using sqlc
- add repositories for ingestion logs and change sets
- run migration on reconciler service startup

Screenshot 2024-12-03 at 14 30 52

### CODE REVIEW

File: diode-server/sqlc.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+version: "2"
```
Comment (leoparente) [inline diode-server/sqlc.yaml:1]: sqlc runs transparently at build time? if not, should we add this to Makefile?

File: diode-server/reconciler/mocks/client.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Code generated by mockery v2.43.2. DO NOT EDIT.
+// Code generated by mockery v2.42.1. DO NOT EDIT.
```
Comment (leoparente) [inline diode-server/reconciler/mocks/client.go]: maybe try to upgrade mockery to latest? but only cosmetic change

Comment (mfiedorowicz) [reply diode-server/reconciler/mocks/client.go]: That's odd, thought I was on the latest hence the change 🤨 Good spot

File: diode-server/sqlc.yaml (hunk 1 above)
Comment (mfiedorowicz) [reply diode-server/sqlc.yaml:1]: sqlc being run locally similarly to mockery, will add recipes for both then

File: diode-server/netboxdiodeplugin/mocks/netboxapi.go [hunk 3]
```diff
@@ -5,9 +5,8 @@ package mocks
 import (
 	context "context"
 
-	mock "github.com/stretchr/testify/mock"
```
Comment (leoparente) [inline diode-server/netboxdiodeplugin/mocks/netboxapi.go:8]: is this change correct?

**Review by leoparente — APPROVED** [review]
Overall, LGMT!

File: diode-server/netboxdiodeplugin/mocks/netboxapi.go (hunk 3 above)
Comment (mfiedorowicz) [reply diode-server/netboxdiodeplugin/mocks/netboxapi.go:8]: That's what mockery generates

File: diode-server/sqlc.yaml (hunk 1 above)
Comment (mfiedorowicz) [reply diode-server/sqlc.yaml:1]: Added following recipes:
```
make install-dev-tools
make gen-mocks
make gen-dbstore
```

Comment (leoparente) [reply diode-server/sqlc.yaml:1]: great! thanks for that

**Pull request merged**